brooklyn-server/camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynAssemblyTemplateInstantiator.java

```diff
@@ -89,6 +90,16 @@ private Application create(AssemblyTemplate template, CampPlatform platform) {
         // first build the children into an empty shell app
         List<EntitySpec<?>> childSpecs = createServiceSpecs(template, platform, loader, encounteredTypeSymbolicNames);
         for (EntitySpec<?> childSpec : childSpecs) {
+            
+            if (Application.class.isAssignableFrom(childSpec.getType())) {
```

Comment:
ah nvm -- it's harmless as it checks if the child might be a wrapper (not what the child wraps), although it's unnecessary as it won't matter if the generic types are wrong and the `canPromote` call will do the check